src/main/java/com/google/cloud/tools/managedcloudsdk/process/CommandExecutor.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.managedcloudsdk.process;
+
+import com.google.cloud.tools.managedcloudsdk.MessageListener;
+import com.google.common.base.Joiner;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+/** Executes a shell command. */
+public class CommandExecutor {
+
+  static final int TIMEOUT_SECONDS = 5;
+
+  private ProcessBuilderFactory processBuilderFactory = new ProcessBuilderFactory();
+  private ExecutorServiceFactory executorServiceFactory = new ExecutorServiceFactory();
+  private MessageListener messageListener;
+  private Map<String, String> environment;
+
+  public CommandExecutor setMessageListener(MessageListener messageListener) {
+    this.messageListener = messageListener;
+    return this;
+  }
+
+  /** Sets the environment variables to run the command with. */
+  public CommandExecutor setEnvironment(Map<String, String> environmentMap) {
+    this.environment = environmentMap;
+    return this;
+  }
+
+  // @VisibleForTesting
```

Comment:
Is it or isn't it? I.e. why the comment?

Reply:
oops I'll fix all those.